protocol/x/clob/memclob/orderbook.go

```diff
@@ -519,26 +477,128 @@ func (m *memclobOpenOrders) mustRemoveOrder(
 	level.LevelOrders.Remove(levelOrder)
 
 	// Decrement the total number of orders for this orderbook.
-	orderbook.TotalOpenOrders--
+	ob.TotalOpenOrders--
 
 	// Edge case: If this removed order was the last order in its
 	// level, we need to remove the level altogether.
 	if level.LevelOrders.Front == nil {
 		delete(levels, subticks)
 		if isBuy {
-			if subticks == orderbook.BestBid {
+			if subticks == ob.BestBid {
 				// Edge case: If this removed level represented the best price level for this side
 				// of the book, we need to find the next best price level.
-				nextBestSubticks, _ := m.findNextBestSubticks(ctx, subticks, orderbook, true)
-				orderbook.BestBid = nextBestSubticks
+				nextBestSubticks, _ := ob.findNextBestSubticks(subticks, true)
+				ob.BestBid = nextBestSubticks
 			}
 		} else {
-			if subticks == orderbook.BestAsk {
+			if subticks == ob.BestAsk {
 				// Edge case: If this removed level represented the best price level for this side
 				// of the book, we need to find the next best price level.
-				nextBestSubticks, _ := m.findNextBestSubticks(ctx, subticks, orderbook, false)
-				orderbook.BestAsk = nextBestSubticks
+				nextBestSubticks, _ := ob.findNextBestSubticks(subticks, false)
+				ob.BestAsk = nextBestSubticks
 			}
 		}
 	}
 }
+
+// get returns the `tilBlock` expiry of an order cancelation and a bool indicating whether the expiry exists.
+func (ob *Orderbook) getCancel(
+	orderId types.OrderId,
+) (
+	tilBlock uint32,
+	exists bool,
+) {
+	tilBlock, exists = ob.orderIdToExpiry[orderId]
+	return tilBlock, exists
```

Comment:
I copied pasted this code. But it needs to force it to return two values instead of just one. if you just `return map[key]` it iwll only be one value.